packages-common/piw-utils/src/utils/PageEditorUtils.ts

```diff
@@ -2,52 +2,73 @@
 /**
  * TODO: Include tests for methods
  */
-import { Properties, Property } from "../typings";
+import { Property, PropertyGroup } from "../typings";
 
 declare type Option<T> = T | undefined;
 
-export function hideProperty<T>(key: keyof T, properties?: Option<Properties>): void {
-    properties?.forEach(property => {
-        property?.properties?.forEach((prop, index, array) => {
-            if (prop.key === key) {
-                array.splice(index, 1);
-            } else {
-                prop.properties?.forEach(propArray => hideProperty(key, propArray));
-            }
-        });
-        hideProperty(key, property.propertyGroups);
-    });
+export function hidePropertyIn(propertyGroups: PropertyGroup[], key: string): void;
+export function hidePropertyIn(
+    propertyGroups: PropertyGroup[],
+    key: string,
```

Comment:
The key is not type safe anymore

Reply:
Yes, but all previous usages of `hideProperty` were using `any` => it wasn't safe already.